megamek/src/megamek/common/Entity.java

```diff
@@ -15473,4 +15460,28 @@ public void setUnitRole(UnitRole role) {
     public UnitRole getRole() {
         return (role == null) ? UnitRole.UNDETERMINED : role;
     }
+
+    /**
+     * Returns the slot in which the given mounted equipment is in its main location. Returns -1 when
+     * the mounted is not in a valid location or cannot be found.
+     *
+     * @param mounted the equipment to look for
+     * @return the (first) slot number that holds the mounted or -1 if none can be found
+     */
+    public int slotNumber(Mounted mounted) {
+        int location = mounted.getLocation();
+        if (location == Entity.LOC_NONE) {
+            return -1;
+        }
+        for (int slot = 0; slot < getNumberOfCriticals(location); slot++) {
```

Comment:
Man, the number of times I see this "loop through all critical slots until we find what we need" code just makes me sad. Wonder if we can include a reverse slot lookup type of thing mapping a Mounted to the list of occupied slots in a particular location. Definitely a project for another PR, though.